src/views/Libraries.vue

```diff
@@ -74,24 +113,95 @@ export default {
       let responses = await Promise.all(promises.map(promise => handlePromise(promise)))
 
       if (responses.every(r => r.successful)) {
-        this.message = `Libraries ${selectedIds.join(',')} successfully deleted`
-        this.showAlert()
+        let keyword = selectedIds.length > 1 ? 'Libraries' : 'Library'
+        this.showAlert(`${keyword} ${selectedIds.join(', ')} successfully deleted`)
       } else {
-        throw responses.map(r => r.errors.message).join(',')
+        throw Error(responses.map(r => r.errors.message).join(','))
       }
     },
-    showAlert () {
-      return this.$refs.alert.show(this.message, 'primary')
+    // Get all the libraries
+    // Provider function used my the bootstrap-vue table component
+    async provider() {
+      this.items = await this.getLibraries()
+    },
+    async getLibraries() {
```

Comment:
Is there room for a bit of refactoring here? Libraries and Samples have almost the same method so maybe it could be abstracted out into a mixin. Also the method is quite long. I don't know whether there is any scope for splitting it up a bit?